Turbolinks/VisitableViewController.swift

```diff
@@ -20,8 +20,14 @@ open class VisitableViewController: UIViewController, Visitable {
 
     fileprivate func installVisitableView() {
         view.addSubview(visitableView)
-        view.addConstraints(NSLayoutConstraint.constraints(withVisualFormat: "H:|[view]|", options: [], metrics: nil, views: [ "view": visitableView! ]))
-        view.addConstraints(NSLayoutConstraint.constraints(withVisualFormat: "V:|[view]|", options: [], metrics: nil, views: [ "view": visitableView! ]))
+        if #available(iOS 11, *) {
+            visitableView.topAnchor.constraint(equalTo: view.safeAreaLayoutGuide.topAnchor).isActive = true
+        } else {
+            visitableView.topAnchor.constraint(equalTo: view.topAnchor).isActive = true
+        }
+        visitableView.bottomAnchor.constraint(equalTo: view.bottomAnchor).isActive = true
+        visitableView.leadingAnchor.constraint(equalTo: view.leadingAnchor).isActive = true
+        visitableView.trailingAnchor.constraint(equalTo: view.trailingAnchor).isActive = true
```

Comment:
We should also use the safe layout guide on the bottom constraint. This helps correctly position buttons that are using `position: fixed;` or `position: absolute;` and "stuck" to the bottom of the screen.

Also, we can wrap everything in a single "activate" call.

```suggestion
        if #available(iOS 11.0, *) {
            NSLayoutConstraint.activate([
                view.leadingAnchor.constraint(equalTo: visitableView.leadingAnchor),
                view.trailingAnchor.constraint(equalTo: visitableView.trailingAnchor),
                view.safeAreaLayoutGuide.topAnchor.constraint(equalTo: visitableView.topAnchor),
                view.safeAreaLayoutGuide.bottomAnchor.constraint(equalTo: visitableView.bottomAnchor)
            ])
        } else {
            view.addConstraints(NSLayoutConstraint.constraints(withVisualFormat: "H:|[view]|", options: [], metrics: nil, views: [ "view": visitableView ]))
            view.addConstraints(NSLayoutConstraint.constraints(withVisualFormat: "V:|[view]|", options: [], metrics: nil, views: [ "view": visitableView ]))
        }
```

Reply:
you right! I faced some issues due to not using safe area layout